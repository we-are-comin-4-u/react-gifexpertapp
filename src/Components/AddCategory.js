import React, { useState } from 'react';
import PropTypes from 'prop-types';
// import { GifExpertApp } from '../GifExpertApp'

export const AddCategory = ({setCategories}) => {

   const [inputValue,setInputValue] = useState('');

   const handleInputChanges = (e) =>{
   // console.log(e.target.value);
      setInputValue(e.target.value);
      // console.log('handleInputChangeswas called sucesfully')

   }

   const handleSubmit = (e) =>{
      e.preventDefault();
      
      // console.log('handleSubmit', inputValue)

       if(inputValue.trim().length > 2){

          setCategories(catego => [ inputValue,...catego,]);
         setInputValue('');
       }

   }

   return (
      
      <form onSubmit={handleSubmit}>
         <p>{inputValue}</p>
         <input 
         type = 'text'
         value={inputValue}
         onChange={handleInputChanges}
         />

      </form>
         
      
   )
}

AddCategory.propTypes = {
   setCategories:PropTypes.func.isRequired
}

export default AddCategory;