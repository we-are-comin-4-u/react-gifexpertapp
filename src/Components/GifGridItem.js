// import { prototype } from '@wojtekmaj/enzyme-adapter-react-17'
import React from 'react';
import PropTypes from 'prop-types';




export const GifGridItem = ({title,url}) => {

   // console.log({id,title,url});
   return (
      <div className = 'card animate__animated animate__bounce'>
         <img src = {url} alt = {title}/>
         <p>{title}</p>
      </div>
   )
}


GifGridItem.propTypes = {
   title: PropTypes.string.isRequired,
   url: PropTypes.string.isRequired,
   
   
}




// import PropTypes from 'prop-types';
