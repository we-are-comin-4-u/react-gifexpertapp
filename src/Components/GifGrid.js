import React from 'react' // { useState,useEffect }
import PropTypes from 'prop-types'

import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifs } from '../helpers/GetGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

   //  const[images,setImages] = useState([])
     const {data:images,loading} = useFetchGifs(category);

   //   console.log(loading);
   //   console.log(data);
   // useEffect(()=>{
   //    getGifs(category).then(setImages);
   // },[category]);

   
      // getGifs();

   return (
      <>
         <h3 className ="animate__fadeIn">{category}</h3>

            { loading && <p className="animate__animated animate__flash">Loading...👩‍💻</p>  }

         
       <div className = "card-grid">
          {/* <h3>{count}</h3> */}
         
            
            
               {
                  images.map(img => ( 
                     <GifGridItem 
                     key = {img.id}//acostumbrate a usar la destructuracion
                        {...img} //esto es un operador expres hay mucho ejemplos en net
                     />  
                  ))

               }
               
      </div> 
      </>
   )
}

GifGrid.propTypes = {
   category: PropTypes.string.isRequired
}