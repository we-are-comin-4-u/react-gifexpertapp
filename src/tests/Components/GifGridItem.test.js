import '@testing-library/jest-dom';

import React from 'react';
import { shallow } from "enzyme";
import {GifGridItem} from '../../Components/GifGridItem';

describe('tests on <GifGridItem/>', () => {

   const title = 'the fresh';
   const url = 'https//localhost/something.jpg';
   const wrapper = shallow(<GifGridItem title={title} url={url}/>);

   test('should to show the right component', () => {
      // const wrapper = shallow(<GifGridItem title={title} url={url}/>);
      expect(wrapper).toMatchSnapshot();
      
   })

   test('should to show a paragraph with the title',()=>{

      const p = wrapper.find('p');
      expect(p.text().trim()).toBe(title);

   })

   test('should to show a imagen with *url* and *alt* of the props' , ()=>{

      const img = wrapper.find('img');
      // console.log(img.props().src);
      expect(img.prop('src')).toBe( url );
      expect(img.prop('alt')).toBe( title );

   })

   test('should to have card animate__animated animate__bounce', () => {

      const div = wrapper.find('div');
      // expect(div.prop('className')).toBe('card animate__animated animate__bounce')
       const className = div.prop('className');

       expect(className.includes('animate__bounce')).toBe(true);
   })
   
   
   
});
