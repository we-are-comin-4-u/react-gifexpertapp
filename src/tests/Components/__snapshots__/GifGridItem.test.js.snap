// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <GifGridItem/> should to show the right component 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="the fresh"
    src="https//localhost/something.jpg"
  />
  <p>
    the fresh
  </p>
</div>
`;
